src/metabase/lib/schema/aggregation.cljc

```diff
@@ -1,47 +1,76 @@
 (ns metabase.lib.schema.aggregation
   (:require
+   [clojure.string :as str]
    [metabase.lib.hierarchy :as lib.hierarchy]
    [metabase.lib.schema.expression :as expression]
    [metabase.lib.schema.mbql-clause :as mbql-clause]
    [metabase.shared.util.i18n :as i18n]
    [metabase.util.malli.registry :as mr]))
 
+;;; `:metric` refs are allowed as aggregations. The definition for `:metric` is in [[metabase.lib.schema.ref]].
+(lib.hierarchy/derive :metric ::aggregation)
+
+;;; Clauses like `:sum` are guaranteed to return a number, right? So rather than return `type.unknown` when their arg
+;;; is something like a `:field` ref without type info, just fall back to `:type/Number`. If we have more specific
+;;; info, we can return a more specific type.
+(lib.hierarchy/derive ::type-is-type-of-first-arg-or-number :lib.type-of/type-is-type-of-first-arg)
+
+(defmethod expression/type-of-method ::type-is-type-of-first-arg-or-number
+  [clause]
+  (let [calculated-type ((get-method expression/type-of-method :lib.type-of/type-is-type-of-first-arg) clause)]
+    (if (= calculated-type ::expression/type.unknown)
+      :type/Number
+      calculated-type)))
```

Comment:
Not strictly required, but type calculation for something like `[:sum {} ...]` should fall back to `:type/Number` rather than `type.unknown`, since we know it will always return a number. Added tests for these improvements